src/Service.Tests/Unittests/ConfigValidationUnitTests.cs

```diff
@@ -1747,5 +1747,140 @@ string[] linkingTargetFields
+                            ]
+                           }
+                         ],
+                        ""rest"":{
+                            ""methods"":" + $"{methods}" + @"
+                         }
+                       }
+                    }
+                }";
+
+            RuntimeConfig runtimeConfig = JsonSerializer.Deserialize<RuntimeConfig>(runtimeConfigString, RuntimeConfig.SerializerOptions);
+
+            // Perform validation on the entity in the config and assert the expected results.
+            if (exceptionExpected)
+            {
+                DataApiBuilderException ex =
+                    Assert.ThrowsException<DataApiBuilderException>(() => RuntimeConfigValidator.ValidateEntityConfiguration(runtimeConfig.Entities));
+                Assert.AreEqual(expectedErrorMessage, ex.Message);
+                Assert.AreEqual(HttpStatusCode.ServiceUnavailable, ex.StatusCode);
+                Assert.AreEqual(DataApiBuilderException.SubStatusCodes.ConfigValidationError, ex.SubStatusCode);
+            }
+            else
+            {
+                RuntimeConfigValidator.ValidateEntityConfiguration(runtimeConfig.Entities);
+            }
+        }
+
+        /// <summary>
+        /// Test to validate that when multiple entities have the same custom rest path configured, we throw an exception.
+        /// </summary>
+        /// <param name="exceptionExpected">Whether an exception is expected as a result of test run.</param>
+        /// <param name="restPathForFirstEntity">Custom rest path to be configured for the first entity.</param>
+        /// <param name="restPathForSecondEntity">Custom rest path to be configured for the second entity.</param>
+        /// <param name="expectedExceptionMessage">The expected exception message.</param>
+        [DataTestMethod]
+        [DataRow(false, "restPathA", "restPathB", DisplayName = "Unique rest paths configured for entities pass config validation.")]
+        [DataRow(false, true, false, DisplayName = "Rest path configured as boolean values for an entities fails config validation.")]
+        [DataRow(true, "restPath", "restPath", "Multiple entities found with same rest path: restPath.",
+            DisplayName = "Duplicate rest paths configures for entities fail config validation.")]
+        [DataRow(true, "", "restPathB", "Entity: EntityA has an empty rest path.",
+            DisplayName = "Empty rest path configured for an entitiy fails config validation.")]
+        [DataRow(true, "restPathA", null, "Entity: EntityB has a null rest path. Accepted value types are: string, boolean.",
+            DisplayName = "NULL rest path configured for an entitiy fails config validation.")]
+        [DataRow(true, "restPathA", 1, $"Entity: EntityB has rest path specified with incorrect data type. Accepted data types are: string, boolean.",
+            DisplayName = "Rest path configured as integer for an entitiy fails config validation.")]
```

Comment:
Do these tests belong here? It seems like these tests would be simpler or achieve the same results by only testing with a single entity and assigning various fault inducing values. e.g. `restPathForFirstEntity` value `restPathA` doesn't affect the outcome of these tests, only the value you provide for `restPathForSecondEntity`